### ISSUE TITLE
Updating to 1.7.2 causes: `Invariant Violation: Element type is invalid`

### ISSUE THREAD

**Issue by mrjones2014** [vue-styleguidist/vue-styleguidist#112]
Usage examples have been updated according to the note on the releases page. Using node version 9.8.0 and npm version 5.7.1.

Full stack trace:
```
Invariant Violation: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
    in ul (created by ListRenderer)
    in ListRenderer (created by Styled(List))
    in Styled(List) (created by Markdown)
    in p (created by ParaRenderer)
    in ParaRenderer (created by Styled(Para))
    in Styled(Para) (created by Markdown)
    in Markdown (created by ReactComponent)
    in div (created by ReactComponentRenderer)
    in div (created by ReactComponentRenderer)
    in ReactComponentRenderer (created by Styled(ReactComponent))
    in Styled(ReactComponent) (created by ReactComponent)
    in ReactComponent (created by Components)
    in div (created by ComponentsRenderer)
    in ComponentsRenderer (created by Components)
    in Components (created by Section)
    in section (created by SectionRenderer)
    in SectionRenderer (created by Styled(Section))
    in Styled(Section) (created by Section)
    in Section (created by Sections)
    in section (created by SectionsRenderer)
    in SectionsRenderer (created by Styled(Sections))
    in Styled(Sections) (created by Sections)
    in Sections (created by StyleGuide)
    in main (created by StyleGuideRenderer)
    in div (created by StyleGuideRenderer)
    in StyleGuideRenderer (created by Styled(StyleGuide))
    in Styled(StyleGuide) (created by StyleGuide)
    in StyleGuide
This may be due to an error in a component you are overriding, or a bug in React Styleguidist.

If you believe this is a bug, please submit an issue.
```

**Comment by rafaesc** [comment]
Hi @mrjones2014,

in which example does it occur? or how can I replicate it?

**Comment by mrjones2014** [comment]
Here is out styleguide.config.js (webpack.common.js is pretty standard):
```js
const webpackConfig = require('./webpack.common.js');
const path = require('path');

module.exports = {
  webpackConfig: webpackConfig,
  title: 'DF Vue Component Documentation',
  template: {
        head: {
            title: 'DF Vue Component Documentation',
            links: [
                {
                    rel: 'stylesheet',
                    href:
                        'https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css'
                },
                {
                    rel: 'stylesheet',
                    href:
                        'https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/css/bootstrap-select.min.css'
                },
                {
                    rel: 'stylesheet',
                    href:
                        'https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css'
                }
            ],
            scripts: [
                { src: 'https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js' },
                { src: 'https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/bootstrap-select.min.js' },
                { src: 'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js' },
            ]
        },
        body: {
            raw: '<div id="rsg-root"></div>'
        }
    },
  showUsage: true,
  showCode: true
};
```

And here is a usage example TagLet.md (note: the final backtick has a space just so it'd show up in this markdown; its correct in the actual file):
```
```js
new Vue({
  data() {
    return {
      allowRemoveAll: true,
      showTagWrapper: true,
      isReadOnly: false,
      enableLink: true,
      sampleRoot: "candy",
      sampleName: "Necco Wafer",
      tags: [],
      defaultTags: [
            {
                root: "plumbus",
                rootId: "42",
                name: "Flarm",
                link: "./plumbus-42.html"
            },
            {
                root: "schmekel",
                name: 25
            }
      ],
      jsonDescriptor:""
    };
  },
  mounted() { 
      this.tags = JSON.parse(JSON.stringify(this.defaultTags));
      this.$nextTick(function() {
            this.updateDescriptor();
      });
  },
  methods: { 
    updateDescriptor() {
      this.jsonDescriptor = JSON.stringify({
      tags: this.$refs.tagLetExample.$props.tags,
      allowRemoveAll: this.$refs.tagLetExample.$props.allowRemoveAll,  
      showTagWrapper: this.$refs.tagLetExample.$props.showTagWrapper,
      readOnly: this.$refs.tagLetExample.$props.readOnly
      }, null, "  ");
    },
    resetComponent() {
        this.tags = JSON.parse(JSON.stringify(this.defaultTags));
        this.allowRemoveAll = true;
        this.showTagWrapper = true;
        this.isReadOnly = false;
        this.enableLink = true;
        this.sampleRoot = "candy";
        this.sampleName = "Necco Wafer";

        this.$nextTick(function() {
            this.updateDescriptor();
        });
    },
    addTag() {
        this.tags.push({root: this.sampleRoot, name: this.sampleName});
        this.updateDescriptor();
    },
    removed() {
      this.updateDescriptor();
    }
  },
  template: `
    <div>
      <div class="form-group form-inline">
        <input class="form-control" type="checkbox" v-model="allowRemoveAll" @change="updateDescriptor">
        <label>&nbsp;Allow Remove All?</label>
        <br />
        <input class="form-control" type="checkbox" v-model="showTagWrapper" @change="updateDescriptor">
        <label>&nbsp;Show Tag Wrapper?</label>
        <br />
        <input class="form-control" type="checkbox" v-model="isReadOnly" @change="updateDescriptor">
        <label>&nbsp;Read Only?</label>
        <br />
        <input class="form-control" type="checkbox" v-model="enableLink" @change="updateDescriptor">
        <label>&nbsp;Enable Link?</label>
        <br />
        <div class="form-group form-inline">
            <label>Root:</label><input type="text" v-model="sampleRoot">
            <label>Name:</label><input type="text" v-model="sampleName">
            <button type="button" class="btn-xs btn-primary" @click="addTag">Add Tag</button>
        </div>
        <br />
        <br />
        <button type="button" class="btn-xs btn-primary" @click="resetComponent">Reset Component</button>
      </div>

      <hr />

      <tag-let ref="tagLetExample"
        :tags="tags"
        :enableLink="enableLink"
        :readOnly="isReadOnly"
        :allowRemoveAll="allowRemoveAll"
        :showTagWrapper="showTagWrapper"
        @onRemove="removed"
        @onAllRemove="removed"
        >
      </tag-let>

      <hr />

      <p>Current TagLet Instance Model. <strong>Note:</strong> only the root and name properties are needed for tag display. The link property is used when the enableLink flag is active. Addtional data properties are optional depending on use case.</p>
      <code style="white-space: pre-wrap;" class="show">{{ jsonDescriptor }}</code>
    </div>
  `
});
` ``
```

**Comment by rafaesc** [comment]
I tried to replicate it here
https://github.com/rafaesc/vue-styleguidist-demo/tree/fix-112

But I couldn't do it. Please send a PR in branch fix-112, with the error to fix it.

Regards

**Comment by elevatebart** [comment]
Is your error fixed with 3.0 ? or should I investigate more ?

**Comment by mrjones2014** [comment]
@elevatebart it seems to have fixed itself

**Comment by elevatebart** [comment]
Neat !!!!
